src/Card.js

```diff
@@ -0,0 +1,313 @@
+      this.flipButton.on('pointerdown',()=>this.startFlip())
+      this.flipButton.on('pointerup',()=>this.flip())
+
+    //card status variables
+    this.cardNumber = cardNumber;
+    this.startFlipClickedDown = false;
+    this.revealed = false;
+    this.spinning = false;
+    this.otherPlayerDragging = false;
+    this.addToHand = false;
+    this.inHand = false;
+    this.playerPickedUp = false;
+    //this.stackNumber should always be the number of the card at the bottom of a given stack.
+    //I.e., card 5 is stacked on card 18, and card 19 is stacked on card 5, all 3 cards will have stackNumber of 18.
+    this.stackNumber = cardNumber
+    //stackOrder is the number of cards from the bottom.
+    //in example above, card18.stackOrder = 1, card5.stackOrder = 2, card19.stackOrder = 3.
+    this.stackOrder = 1
+
+    //overwrite phaser's bullshit toJSON implementation
+    this.toJSON = () => {
+      return {
+        cardNumber: this.cardNumber,
+        x: this.x,
+        y: this.y,
+        rotation: this.rotation,
+        revealed: this.revealed,
+        velocity: this.body.velocity,
+        otherPlayerDragging: this.otherPlayerDragging,
+        inHand: this.inHand,
+        player: this.playerNumber,
+        stackNumber: this.stackNumber,
+        stackOrder: this.stackOrder,
+        depth: this.depth
+      }
+    }
+  }
+
+  getCardsInStack(stackNumber = this.stackNumber) {
+    return Object.keys(this.gameState.cards)
+    .map( key => this.gameState.cards[key])
+    .filter( card => card.stackNumber === stackNumber)
+    .sort( (a,b) => a.stackOrder - b.stackOrder)
+  }
+
+  giveNextStackNumber(stackNumber = this.stackNumber) {
+    return this.getCardsInStack(stackNumber).length + 1
+  }
+
+  getClosestCardNotInStack() {
+    const roomCards = Object.keys(this.gameState.cards).filter(key => Number(key) !== this.cardNumber && this.gameState.cards[key].stackNumber !== this.stackNumber).map( key => this.gameState.cards[key])
+    return this.scene.physics.closest(this, roomCards)
+  }
+
+  dragStart() {
+
+    if(!this.otherPlayerDragging) this.playerPickedUp = true
+  }
+
+  drag (ptr, dragX, dragY, dragStack = false) {
+    if (!dragStack) {
+      this.stackNumber = this.cardNumber;
+      this.stackOrder = 1;
+    }
+
+    if(!this.playerPickedUp) return
+    this.setDepth(activeDepth)
+
+    //if a card is in your hand
+    if (this.inHand) {
+      //lock it to the bottom
+      switch(this.playerNumber) {
+        case 2:
+          this.y = dragY < 0 ? 0 : dragY > 800 ? 800 : dragY;
+          this.x = canvasWidth + inHandAdjustment;
+          break;
+        case (3):
+          this.x = dragX < 0 ? 0 : dragX > 800 ? 800 : dragX;
+          this.y = 0 - inHandAdjustment;
+          break;
+        case 4:
+          this.y = dragY < 0 ? 0 : dragY > 800 ? 800 : dragY;
+          this.x = 0 - inHandAdjustment;
+          break;
+        default:
+          this.x = dragX < 0 ? 0 : dragX > 800 ? 800 : dragX;
```

Comment:
You can probably break this out into its own function since you use it multiple times, something like:
```js
putInBounds(location) {
  if(location > 800) return 800;
  else if(location < 800) return 0;
  else return location
}
```

Furthermore, right now 800 is a "magic number" - or a value that only works because the developers happened to hard code it in the way it is. This can cause bugs in the long run and make the code harder to refactor, I'd think about moving this outside of this file into some sort of global state representing the board constants / it'd be better to use the board width and height as listed in your constants.js file